Build failed on MinGW (mingw-w64) nng 1.1.1

## NNG & Platform details.

* nng version: 1.1.1
* Debian 4.19.13
* cmake version 3.7.2

tried with nng-rs (Rust), got the same error, reproduced using Cmake directly on a clone of nng repo.
## Expected Behavior
Compilation successful using mingw to target windows with the following toolchain or ```cargo build --target x86_64-pc-windows-gnu```

```
set(CMAKE_SYSTEM_NAME Windows)
set(TOOLCHAIN_PREFIX x86_64-w64-mingw32)

# cross compilers to use for C and C++
set(CMAKE_C_COMPILER ${TOOLCHAIN_PREFIX}-gcc)
set(CMAKE_CXX_COMPILER ${TOOLCHAIN_PREFIX}-g++)
set(CMAKE_Fortran_COMPILER ${TOOLCHAIN_PREFIX}-gfortran)
set(CMAKE_RC_COMPILER ${TOOLCHAIN_PREFIX}-windres)

# target environment on the build host system
#   set 1st to dir with the cross compiler's C/C++ headers/libs
set(CMAKE_FIND_ROOT_PATH /usr/${TOOLCHAIN_PREFIX})

# modify default behavior of FIND_XXX() commands to
# search for headers/libs in the target environment and
# search for programs in the build host environment
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
```
## Actual Behavior
Compilation fails because of undefined reference to `InterlockedAddNoFence64'

### C
####  cmake .. -DNNG_TOOLS=ON -DCMAKE_TOOLCHAIN_FILE=../toolchain
```
root@0d4409fe05f9:/# cd /nng/build && cmake .. -DCMAKE_TOOLCHAIN_FILE=../toolchain
-- The C compiler identification is GNU 6.3.0
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found UnixCommands: /bin/bash  
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for InitializeConditionVariable
-- Looking for InitializeConditionVariable - found
-- Looking for snprintf
-- Looking for snprintf - found
-- Looking for strlcat
-- Looking for strlcat - not found
-- Looking for strlcpy
-- Looking for strlcpy - not found
-- Looking for strnlen
-- Looking for strnlen - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strncasecmp
-- Looking for strncasecmp - found
-- Building static libs
-- Configuring done
-- Generating done
-- Build files have been written to: /nng/build
```

#### make
```
[ 98%] Building C object tools/nngcat/CMakeFiles/nngcat.dir/nngcat.c.obj
[100%] Linking C executable nngcat.exe
../../libnng.a(win_thread.c.obj):win_thread.c:(.text+0x2a8): undefined reference to `InterlockedAddNoFence64'
../../libnng.a(win_thread.c.obj):win_thread.c:(.text+0x2d5): undefined reference to `InterlockedAddNoFence64'
collect2: error: ld returned 1 exit status
tools/nngcat/CMakeFiles/nngcat.dir/build.make:98: recipe for target 'tools/nngcat/nngcat.exe' failed
```

### Rust

#### cargo build --target x86_64-pc-windows-gnu
```
error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1   
[...]
note: /nng_compile_error/target/x86_64-pc-windows-gnu/debug/deps/libnng_sys-a8ae54a6ecfd733b.rlib(win_thread.c.obj): In function `nni_atomic_inc64':                                                                                      
          /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/nng-sys-0.1.3/libnng/src/platform/windows/win_thread.c:127: undefined reference to `InterlockedAddNoFence64'                                                              
          /nng_compile_error/target/x86_64-pc-windows-gnu/debug/deps/libnng_sys-a8ae54a6ecfd733b.rlib(win_thread.c.obj): In function `nni_atomic_dec64':                                                                                      
          /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/nng-sys-0.1.3/libnng/src/platform/windows/win_thread.c:134: undefined reference to `InterlockedAddNoFence64'                                                              
          collect2: error: ld returned 1 exit status                    
```

## Steps to Reproduce
### C
```
docker run --rm -ti -w /nng/build forgottenbeast/nng_compile_error bash;
# once inside the docker container
cmake .. -DNNG_TOOLS=ON -DCMAKE_TOOLCHAIN_FILE=../toolchain && make;
```

### Rust
```
docker run --rm -ti forgottenbeast/nng_compile_error bash;
# once inside the docker container
cargo build --target x86_64-pc-windows-gnu
```

--- a/src/platform/windows/win_thread.c
+++ b/src/platform/windows/win_thread.c
@@ -14,6 +14,11 @@
 
 #ifdef NNG_PLATFORM_WINDOWS
 
+//mingw does not define InterlockedAddNoFence64, use the mingw equivelent
+#ifdef __MINGW32__ || __MINGW64__
+#define InterlockedAddNoFence64(a,b) __atomic_add_fetch(a,b,__ATOMIC_RELAXED)
+#endif
+
 #include <stdlib.h>
 
 void *

